crypto_secretstream/src/stream.rs

```diff
@@ -0,0 +1,294 @@
+use std::convert::TryFrom;
+
+use aead::{AeadCore, AeadInPlace};
+use chacha20::{
+    cipher::{
+        consts::U16,
+        generic_array::{
+            functional::FunctionalSequence,
+            sequence::{Concat, Split},
+            GenericArray,
+        },
+        NewCipher, StreamCipher, StreamCipherSeek,
+    },
+    hchacha, ChaCha20, R20,
+};
+use poly1305::{
+    universal_hash::{NewUniversalHash, Output, UniversalHash},
+    Poly1305,
+};
+use rand_core::{CryptoRng, RngCore};
+use subtle::ConstantTimeEq;
+
+use super::nonce::Nonce;
+use crate::{header::Header, Key, Tag};
+
+const MAC_BLOCK_SIZE: usize = 16;
+const TAG_BLOCK_SIZE: usize = 16 * 4;
+
+type Result<T> = std::result::Result<T, aead::Error>;
```

Comment:
I just realized we don't define this alias upstream in the `aead` crate. I'll go ahead and add it.

Reply:
Adding it here: https://github.com/RustCrypto/traits/pull/725

You should be able to use `aead::Result` as soon as I cut a release.